streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -1012,11 +1108,84 @@ void to(final String topic,
      * @param <VR>                     the value type of the result stream
      * @return a {@code KStream} that contains records with unmodified key and new values (possibly of different type)
      * @see #mapValues(ValueMapper)
+     * @see #mapValues(ValueMapperWithKey)
      * @see #transform(TransformerSupplier, String...)
      */
     <VR> KStream<K, VR> transformValues(final ValueTransformerSupplier<? super V, ? extends VR> valueTransformerSupplier,
                                         final String... stateStoreNames);
 
+    /**
+     * Transform the value of each input record into a new value (with possible new type) of the output record.
+     * A {@link ValueTransformerWithKey} (provided by the given {@link ValueTransformerWithKeySupplier}) is applies to each input
+     * record value and computes a new value for it.
+     * Thus, an input record {@code <K,V>} can be transformed into an output record {@code <K:V'>}.
+     * This is a stateful record-by-record operation (cf. {@link #mapValues(ValueMapper)}).
+     * Furthermore, via {@link org.apache.kafka.streams.processor.Punctuator#punctuate(long)} the processing progress can be observed and additional
+     * periodic actions get be performed.
+     * <p>
+     * In order to assign a state, the state must be created and registered beforehand:
+     * <pre>{@code
+     * // create store
+     * StateStoreSupplier myStore = Stores.create("myValueTransformState")
```

Comment:
nit: the `StateStoreSupplier` example is deprecated API maybe change to something like
```
StoreBuilder<KeyValueStore<String,String>> keyValueStoreBuilder = 
Stores.keyValueStoreBuilder(Stores.persistentKeyValueStore("myValueTransformState"),
                             Serdes.String(),
                             Serdes.String());

builder.addStateStore(keyValueStore);
```

Reply:
Thanks a lot for the feedback. Fixed.